src/server/tran_server.cpp

```diff
@@ -318,19 +322,26 @@ tran_server::get_boot_info_from_page_server ()
 }
 
 int
-tran_server::connect_to_page_server (connection_handler &conn_handler, const cubcomm::node &node, const char *db_name)
+tran_server::connection_handler::connect (const cubcomm::node &node)
 {
-  auto ps_conn_error_lambda = [&node] ()
+  auto ps_conn_error_lambda = [&node, this] ()
```

Comment:
Nit.
To formally ensure that the lock is in effect.
```suggestion
  auto ps_conn_error_lambda = [&node, this] (const std::lock_guard<std::shared_mutex>&)
```
I know it makes the code ugly. :)

Reply:
Seems better to me. Thanks.